queue/queued_channel.go

```diff
@@ -18,7 +18,15 @@ type QueuedChannel[T any] struct {
 	closed atomicBool // Should use atomic.Bool once we use Go 1.19!
 }
 
-func NewQueuedChannel[T any](chanBufferSize, queueCapacity int) *QueuedChannel[T] {
+type PanicHandler interface {
```

Comment:
Can we have this in a different package/folder. I feel it would be better to have this self contained rather than being part of the queue.